pkg/token/token_test.go

```diff
@@ -38,156 +41,126 @@ func generateTestKey() (*rsa.PrivateKey, error) {
 }
 
 func TestToken(t *testing.T) {
-	issuer := "issuer"
-
-	key, err := generateTestKey()
-	if err != nil {
-		t.Errorf("error generating keys: %v", err)
-	}
-
-	tokenData := &Payload{
-		Operation: OperationUpload,
-		Name:      "fakepvc",
-		Namespace: "fakenamespace",
-		Resource: metav1.GroupVersionResource{
-			Group:    "",
-			Version:  "v1",
-			Resource: "persistentvolumeclaims",
-		},
-	}
-
-	g := NewGenerator(issuer, key, 5*time.Minute)
-
-	signedToken, err := g.Generate(tokenData)
-
-	if err != nil {
-		t.Errorf("unable to generate token: %v", err)
-	}
-
-	validator := NewValidator(issuer, &key.PublicKey, 0)
-
-	payload, err := validator.Validate(signedToken)
-
-	if err != nil {
-		t.Errorf("unable to verify token: %v", err)
-	}
-
-	if !reflect.DeepEqual(tokenData, payload) {
-		t.Errorf("invalid token payload")
-	}
+	RegisterFailHandler(Fail)
```

Comment:
In general we try to have a single 'suite' file that has the 'Testing.T' in it. We name it: _suite_test.go with the package here being apiserver. It also allows us to set things like logger in the before suite. A decent example is in pkg/controller/controller_suite_test.go